src/test/java/bot/utils/NameProcessorTest.java

```diff
@@ -0,0 +1,304 @@
+package bot.utils;
+
+import org.junit.Test;
+
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.List;
+
+import static org.junit.Assert.*;
+
+public class NameProcessorTest {
+
+    private String input;
+    private List<String> expected;
+    private List<String> output;
+
+    private void runTest() {
+        output = NameProcessor.generateAcceptedNames(input);
+        Collections.sort(expected);
+        Collections.sort(output);
+        assertEquals(output.containsAll(expected), expected.containsAll(output));
+//        assertEquals(expected, output);
+    }
+
+    @Test
+    public void givenSampleName_expectArrayOfStrings001() {
+        input = "Al.Zieleniecka";
+        expected = Arrays.asList("al zieleniecka", "aleja zieleniecka");
+        runTest();
+    }
+
+    @Test
+    public void givenSampleName_expectArrayOfStrings002() {
+        input = "MURANÓW";
+        expected = Arrays.asList("muranow");
+        runTest();
+    }
+
+    @Test
+    public void givenSampleName_expectArrayOfStrings003() {
+        input = "Marymont-Potok";
+        expected = Arrays.asList("marymont potok");
+        runTest();
+    }
+
+    @Test
+    public void givenSampleName_expectArrayOfStrings004() {
+        input = "Metro Ratusz Arsenał";
+        expected = Arrays.asList("metro ratusz arsenal", "ratusz arsenal");
+        runTest();
+    }
+
+    @Test
+    public void givenSampleName_expectArrayOfStrings005() {
+        input = "Małcużyńskiego";
+        expected = Arrays.asList("malcuzynskiego");
+        runTest();
+    }
+
+    @Test
+    public void givenSampleName_expectArrayOfStrings006() {
+        input = "Cm.Żydowski";
+        expected = Arrays.asList("cm zydowski", "cmentarz zydowski");
+        runTest();
+    }
+
+    @Test
+    public void givenSampleName_expectArrayOfStrings007() {
+        input = "Al.Piłsudskiego";
+        expected = Arrays.asList("al pilsudskiego", "aleja pilsudskiego");
+        runTest();
+    }
+
+    @Test
+    public void givenSampleName_expectArrayOfStrings008() {
+        input = "Grupy AK \"Północ\"";
+        expected = Arrays.asList("grupy ak polnoc");
+        runTest();
+    }
+
+    @Test
+    public void givenSampleName_expectArrayOfStrings009() {
+        input = "Al.Jana Pawła II";
+        expected = Arrays.asList("al jana pawla ii, al jana pawla 2, aleja jana pawla 2, aleja jana pawla ii");
```

Comment:
This list contains exactly 1 `String`, not 4 :)

Reply:
xDDDDDDDDDDDDDD